client-network/src/network_ops/default_network_ops_client.rs

```diff
@@ -8,41 +10,77 @@ use chain_core::tx::data::attribute::TxAttributes;
 use chain_core::tx::data::input::TxoPointer;
 use chain_core::tx::data::output::TxOut;
 use chain_core::tx::{TransactionId, TxAux};
+use client_common::tendermint::Client;
 use client_common::{Error, ErrorKind, Result};
 use client_core::signer::Signer;
 use client_core::UnspentTransactions;
 use client_core::WalletClient;
+use parity_codec::Decode;
 use secstr::SecUtf8;
-
 /// Default implementation of `NetworkOpsClient`
-pub struct DefaultNetworkOpsClient<'a, W, S>
+pub struct DefaultNetworkOpsClient<'a, W, S, C>
 where
     W: WalletClient,
     S: Signer,
+    C: Client,
 {
     wallet_client: &'a W,
     signer: &'a S,
+    client: &'a C,
 }
 
-impl<'a, W, S> DefaultNetworkOpsClient<'a, W, S>
+impl<'a, W, S, C> DefaultNetworkOpsClient<'a, W, S, C>
 where
     W: WalletClient,
     S: Signer,
+    C: Client,
 {
     /// Creates a new instance of `DefaultNetworkOpsClient`
-    pub fn new(wallet_client: &'a W, signer: &'a S) -> Self {
+    pub fn new(wallet_client: &'a W, signer: &'a S, client: &'a C) -> Self {
         Self {
             wallet_client,
             signer,
+            client,
         }
     }
+
+    /// Get account info
+    pub fn get_account(&self, staked_state_address: &[u8]) -> Result<StakedState> {
+        self.client
+            .query("account", hex::encode(staked_state_address).as_str())
+            .map(|x| x.response.value)
+            .and_then(|value| match base64::decode(value.as_bytes()) {
+                Ok(a) => Ok(a),
+                Err(_b) => Err(Error::from(ErrorKind::RpcError)),
+            })
+            .and_then(|data| match StakedState::decode(&mut data.as_slice()) {
+                Some(a) => Ok(a),
+                None => Err(Error::from(ErrorKind::RpcError)),
+            })
+    }
 }
 
-impl<'a, W, S> NetworkOpsClient for DefaultNetworkOpsClient<'a, W, S>
+impl<'a, W, S, C> NetworkOpsClient for DefaultNetworkOpsClient<'a, W, S, C>
 where
     W: WalletClient,
     S: Signer,
+    C: Client,
 {
+    fn get_staked_state_account(
+        &self,
+        to_staked_account: StakedStateAddress,
+    ) -> Result<StakedState> {
+        match to_staked_account {
+            StakedStateAddress::BasicRedeem(a) => self.get_account(&a.0),
+        }
+    }
+
+    fn get_staked_state_nonce(&self, to_staked_account: StakedStateAddress) -> Result<Nonce> {
```

Comment:
as suggested by @devashishdxt , this method is probably not necessary -- nonce can be retrieved in one line:
```
let nonce = self.get_staked_state_account(to_staked_account)?.nonce;
```

Reply:
got it